docs/api/dataset/index.rst

```diff
@@ -0,0 +1,30 @@
+.. _dataset_api :
+
+qcodes.dataset
+==============
+
+
+.. autosummary::
+
+    qcodes.dataset
+    qcodes.dataset.measurements
+    qcodes.dataset.plotting
+    qcodes.dataset.data_set
+    qcodes.dataset.database_extract_runs
+    qcodes.dataset.legacy_import
+    qcodes.dataset.param_spec
```

Comment:
as per some discussions last year, i still don't think that ParamSpec should be exposed to the users. at least for now.